internal/source/kubernetes/source.go

```diff
@@ -184,78 +241,119 @@ func consumeEvents(ctx context.Context, s Source) {
 
-	// Skip older events
-	if !e.TimeStamp.IsZero() && e.TimeStamp.Before(s.startTime) {
-		s.logger.Debug("Skipping older events")
-		return
-	}
+	return func(ctx context.Context, e event.Event, sources, updateDiffs []string) {
+		globalLogger.Debugf("Processing %s to %s/%v in %s namespace", e.Type, e.Resource, e.Name, e.Namespace)
 
-	// Check for significant Update Events in objects
-	if e.Type == config.UpdateEvent && len(updateDiffs) > 0 {
-		e.Messages = append(e.Messages, updateDiffs...)
-	}
+		// Skip older events
+		if !e.TimeStamp.IsZero() && e.TimeStamp.Before(s.bgProcessor.StartTime()) {
+			globalLogger.Debug("Skipping older event...")
+			return
+		}
 
-	// Filter events
-	e = s.filterEngine.Run(ctx, e)
-	if e.Skip {
-		s.logger.Debugf("Skipping event: %#v", e)
-		return
-	}
+		// Check for significant Update Events in objects
+		if e.Type == config.UpdateEvent && len(updateDiffs) > 0 {
+			e.Messages = append(e.Messages, updateDiffs...)
+		}
 
-	if len(e.Kind) <= 0 {
-		s.logger.Warn("sendEvent received e with Kind nil. Hence skipping.")
-		return
-	}
+		errs := multierror.New()
+		for _, sourceKey := range sources {
+			eventCopy := e
 
-	recRunner, recCfg := s.recommFactory.New(s.config)
-	err := recRunner.Do(ctx, &e)
-	if err != nil {
-		s.logger.Errorf("while running recommendations: %w", err)
-		return
-	}
+			srcCfg, ok := s.configStore.Get(sourceKey)
+			if !ok {
+				errs = multierror.Append(errs, fmt.Errorf("source with key %q not found", sourceKey))
+			}
 
-	if recommendation.ShouldIgnoreEvent(&recCfg, e) {
-		s.logger.Debugf("Skipping event as it is related to recommendation informers and doesn't have any recommendations: %#v", e)
-		return
-	}
+			if srcCfg.ActiveSourceConfig == nil {
+				globalLogger.Errorf("ActiveSourceConfig not found for source %s", srcCfg.name)
+				continue
+			}
 
-	msg, err := s.messageBuilder.FromEvent(e, s.config.ExtraButtons)
-	if err != nil {
-		s.logger.Errorf("while rendering message from event: %w", err)
-		return
+			setClusterName(srcCfg.clusterName, &eventCopy)
+
+			// Filter events
+			e = srcCfg.filterEngine.Run(ctx, eventCopy)
+			if e.Skip {
+				srcCfg.logger.Debugf("Skipping event: %#v", eventCopy)
+				continue
+			}
+
+			if len(e.Kind) <= 0 {
+				srcCfg.logger.Warn("sendEvent received e with Kind nil. Hence skipping.")
+				continue
+			}
+
+			recRunner, recCfg := srcCfg.recommFactory.New(srcCfg.cfg)
+			err := recRunner.Do(ctx, &eventCopy)
+			if err != nil {
+				errs = multierror.Append(errs, fmt.Errorf("while running recommendation: %w", err))
+				continue
+			}
+
+			if recommendation.ShouldIgnoreEvent(&recCfg, eventCopy) {
+				srcCfg.logger.Debugf("Skipping event as it is related to recommendation informers and doesn't have any recommendations: %#v", e)
+				continue
+			}
+
+			msg, err := srcCfg.messageBuilder.FromEvent(eventCopy, srcCfg.cfg.ExtraButtons)
+			if err != nil {
+				srcCfg.logger.Errorf("while rendering message from event: %w", err)
```

Comment:
why here we just log it and not append to `errs`? on the other hand,  in line 312 we append it to `errs` but we don't log it. Please pick unified approach.